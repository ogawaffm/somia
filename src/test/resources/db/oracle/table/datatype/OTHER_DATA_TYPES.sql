BEGIN
    EXECUTE IMMEDIATE 'DROP TABLE TEST.OTHER_DATA_TYPES';
EXCEPTION
    WHEN OTHERS THEN
        IF SQLCODE != -942 THEN
            RAISE;
        END IF;
END;
/
CREATE TABLE TEST.OTHER_DATA_TYPES
(
    ROWID_COL        ROWID,

    UROWID_COL       UROWID,
    UROWID_1_COL     UROWID(1),
    UROWID_4000_COL  UROWID(4000),

    BFILE_COL        BFILE,

    /* moved from CHAR_DATA_TYPES.sql because LONG and LONG VARCHAR are not
       allowed in the same table                                            */
    LONG_VARCHAR_COL LONG VARCHAR,

    BOOLEAN_COL      BOOLEAN,
    JSON_COL         JSON

)