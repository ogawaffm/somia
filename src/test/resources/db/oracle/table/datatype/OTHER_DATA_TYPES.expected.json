{
  "name": "OTHER_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "ROWID_COL",
      "type": {
        "name": "ROWID",
        "typeNumber": -8,
        "className": "java.sql.RowId",
        "precision": 10,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "UROWID_COL",
      "type": {
        "name": "UROWID",
        "typeNumber": -8,
        "className": "java.sql.RowId",
        "precision": 4000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "UROWID_1_COL",
      "type": {
        "name": "UROWID",
        "typeNumber": -8,
        "className": "java.sql.RowId",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "UROWID_4000_COL",
      "type": {
        "name": "UROWID",
        "typeNumber": -8,
        "className": "java.sql.RowId",
        "precision": 4000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "BFILE_COL",
      "type": {
        "name": "BFILE",
        "typeNumber": -13,
        "className": "java.lang.Object",
        "precision": 530,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "LONG_VARCHAR_COL",
      "type": {
        "name": "LONG",
        "typeNumber": -1,
        "className": "java.lang.String",
        "precision": 0,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 65535,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "BOOLEAN_COL",
      "type": {
        "name": "BOOLEAN",
        "typeNumber": 16,
        "className": "java.lang.Boolean",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 5,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "JSON_COL",
      "type": {
        "name": "JSON",
        "typeNumber": 2016,
        "className": "java.lang.Object",
        "precision": 8200,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": true,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}